### ISSUE TITLE
[Question] Loading CSS

### ISSUE THREAD

**Issue by VentyCZ** [AngleSharp/AngleSharp#303]
Hey there,
I'm happy that I got to know this project - convenient package !
Now, to the issue:
I'm trying to load really simple HTML with some styling defined in the HEAD - right in the same document, not in external file using link. And the problem is that the rules from the STYLE tags are not reflected on the elements itself, but when the style property of the elements itself is used, the style properties from the style property is reflected on the Style property of the element.

How can I enable loading this info into the elements ? Is there such functionality ? Or should I get the rules and apply them manually by setting the style properties on these elements (which is an option, but odd) ?

I'm initializing the AS like below

``` csharp
var parser = new HtmlParser(Configuration.Default.WithCss());
var DOM = parser.Parse(source);
```

Thanks for response.

**Comment by FlorianRappl** [comment]
AngleSharp is used like the browser. The `Style` property only contains information about the inline style, i.e., an `ICssStyleDeclaration` obtained by parsing the `style` attribute of the corresponding element.

What you want to go for is `GetComputedStyle` available on the `IWindow`, but be careful - this feature is not completely ready yet and only performs trivial computations to obtain an `ICssStyleDeclaration` of the given element using all _active_ stylesheets (linked and inline) / style information.

Maybe I'll add a handy helper that is used like

``` cs
var style = element.ComputeCurrentStyle();
```

instead of the current, explicit version

``` cs
var window = element.Owner.DefaultView;
var style = window.GetComputedStyle(element);
```

**Comment by VentyCZ** [comment]
I'm reporting after testing it & I'm really happy with the results !!! Thank you very much.
Can you explain more deeply or give example of the issues that can come up by using this method? How careful should I be ?

**Comment by FlorianRappl** [comment]
What I mean with issues is that it does not do any viewport-related style computations. It only checks the precedence by taking into consideration the assigned rules (priority) and walking down the tree - carrying potential inheritance of CSS rules.

So besides the obvious deficit of not calculating the style (and being unable to use any viewport-related information) it may use wrong rules for applying CSS inheritance. Currently, the model seems to work in trivial examples, but as its not fully adjusted to the spec there may be subtle issues. I'll need to tackle them soon (before v1).

I don't know how careful you should be. I would say for simple things it should be reliable, but for complicated things I would go for a few tests first. But that fully dependents on the scenario you have on your mind.

**Comment by FlorianRappl** [comment]
Included, see #306.

**Comment by VentyCZ** [comment]
Awesome news ! Thanks for info & update.
